### DISCUSSION TITLE
Hot Reload & Community Contributions

### DISCUSSION THREAD

**Discussion by MeikTranel** [dotnet-foundation/Home#65] [General Discussions]
I am sure I don't have to explain the hot reload / dotnet watch situation since probably most of the community heard of it by now. For those out of the loop:
* the problem: https://github.com/dotnet/sdk/issues/22247
* the resolution: https://devblogs.microsoft.com/dotnet/net-hot-reload-support-via-cli/

Now I *specifically* don't want to relitigate the hot reload discussion itself since it's been resolved and the core issue is something I feel is either only tangentially related to the dotnet foundation or being discussed in more appropriate contexts in other discussion threads around here.

But there remains one issue that I feel hasn't been talked about yet (the amount of comments just makes it impossible for me to be sure about this):

When Microsoft employed maintainers (who we as the dotnet foundation entrusted with keeping the status as maintainers of the dotnet core platform) removed the hot reload code from the SDK repo it didn't just remove the committed contributions of Microsoft engineers but also from open source contributors. I've had contact with one of those contributors to get a sense of how they feel about that and to get some facts on what the contribution was in terms of work. 

I'm sure at least some of you who read this have in the past contributed in some way or another to one of the Microsoft maintained dotnet codebases and thus have some experience what writing up an issue or a PR there entails. In the case of the individual I contacted it was an estimated one or two hours of work.

Small tangent: I also asked them if they signed a CLA since the PR didn't have a CLA bot comment which would dress this issue in a much more serious gown - but in their case the CLA was signed; the bot just didn't comment.

### Coming back to my point: 

_**In what is reported to be a move by a CVP to better the monetization of VS, Microsoft removed without coordinating with the dotnet foundation code that was written by open source contributors at a time where they could reasonably assume it would help improve the open source platform dotnet by landing in the next version of the dotnet SDK since that was the commitment at the time.**_

### So the questions are:

* **Will there ever be coordination between Microsoft & the dotnet foundation on direction of the platform?**
* **Do we accept Microsoft deleting community contributions of which the ownership was transferred to the dotnet foundation via the CLA?**
* **How do we gain back trust of those that intended to contribute but may be deterred right now by this move?**

P.S.: One thing I am unsure of that I would really appreciate if someone can answer this for me: Is the VS implementation of hot reload in any way dependent on (or a copy of ) the code that was removed?

**Comment by MSDN-WhiteKnight** [comment]
>Is the VS implementation of hot reload in any way dependent on (or a copy of ) the code that was removed?

It is the opposite way, the dotnet watch implementation depends on VS implementation. It was updated to enable dotnet watch in this PR: https://github.com/dotnet/roslyn/pull/51967

**Reply by MeikTranel** [reply]
That's not VS though. That is roslyn exposing the APIs to VS and dotnet watch. Looking at that code the list of contributions by non-microsoft people goes even further.

**Comment by svick** [comment]
> Do we accept Microsoft deleting community contributions of which the ownership was transferred to the dotnet foundation via the CLA?

I think we have to. Code can be deleted for a variety of reasons, most of which are completely valid. The real problem here is that MS violated the community's trust (whether it was malicious or not), but I don't think creating bureaucratic hurdles can help with building the trust back up. This rule wouldn't really limit them from being malicious in the future (e.g. they could disable the code instead of deleting it, which was mentioned as somehow better in the linked blog post) and could actually make MS less open (if they can't delete potentially problematic code, they'll be more wary of accepting it in the first place). So I don't think a rule like this one would help anything.
